agreement/asyncVoteVerifier.go

```diff
@@ -129,7 +130,10 @@ func (avv *AsyncVoteVerifier) verifyVote(verctx context.Context, l LedgerReader,
 	default:
 		// if we're done while waiting for room in the requests channel, don't queue the request
 		req := asyncVerifyVoteRequest{ctx: verctx, l: l, uv: &uv, index: index, message: message, out: out}
-		avv.backlogExecPool.EnqueueBacklog(avv.ctx, avv.executeVoteVerification, req, avv.execpoolOut)
+		avv.wg.Add(1)
+		if avv.backlogExecPool.EnqueueBacklog(avv.ctx, avv.executeVoteVerification, req, avv.execpoolOut) != nil {
+			avv.wg.Done()
```

Comment:
Can you add a comment here explaining why we're calling .Done() only when this returns nil?

Reply:
sure.